common/cmake/devices.cmake

```diff
@@ -346,7 +355,7 @@ function(DEFINE_DEVICE_TYPE)
   set(ARCH_SCHEMA ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/fpga_architecture.xsd)
   xml_lint(
     NAME ${DEFINE_DEVICE_TYPE_ARCH}_${DEFINE_DEVICE_TYPE_DEVICE_TYPE}_arch_lint
-    FILE ${FINAL_OUTPUT}
+    FILE ${FINAL_FILE}
```

Comment:
This should be FINAL_OUTPUT?  And xml_lint should invoke get_file_location + append_file_dependency?

Reply:
Are you suggesting that I modify `xml_lint()` to change how the FILE parameter is expected?

Reply:
``xml_lint()`` is currently broken if you attempt to invoke it from a directory other than the directory in which the add_custom_command was defined.  If ``xml_lint()`` is invoked after the file is already generated, everything works out, but it doesn't model the target dependency correctly.

Reply:
This seems out of scope of this change. FINAL_FILE is the full path so bypasses the problem you mentioned, but I agree this should be addressed. Creating a separate issue.